src/Common/src/TypeSystem/Ecma/EcmaSignatureParser.cs

```diff
@@ -288,5 +288,46 @@ public TypeDesc[] ParseMethodSpecSignature()
             }
             return arguments;
         }
+
+        private NativeType ReadAndValidateNativeType()
+        {
+            byte nextByte = _reader.ReadByte();
+            NativeType nativeType = NativeType.NATIVE_TYPE_INVALID;
+            foreach (NativeType type in Enum.GetValues(typeof(NativeType)))
```

Comment:
We avoid use of reflection in the type system because it's not low level enough. The type loader is also used at runtime where it's layered below reflection. Also, reflection is just slow.

Do we need this sort of validation? I can image other reasons why the native type could be invalid (e.g. I can put `NATIVE_TYPE_LPWSTR` on a `bool` parameter). Why not just validate at the time the information is used (when we have more context).